File: tsPaymentApi/src/services/mongo/MongoDBService.ts
import { MongoClient, Db } from 'mongodb';

export abstract class MongoDBService {
    private readonly client: MongoClient;
    protected readonly db: Db;

    constructor(databaseName: string = 'Store') {
        const mongoUri = process.env.MONGO_URI || 'mongodb://localhost:27017/';
        this.client = new MongoClient(mongoUri);
        this.db = this.client.db(databaseName);
    }

    protected async connect(): Promise<void> {
        try {
            await this.client.connect();
        } catch (error) {
            console.error('Failed to connect to MongoDB:', error);
            throw error;
        }
    }

    protected async disconnect(): Promise<void> {
        try {
            await this.client.close();
        } catch (error) {
            console.error('Failed to disconnect from MongoDB:', error);
            throw error;
        }
    }
} 

File: py_payment_api/app/services/mongo/mongo_db_service.py
from pymongo import MongoClient
import os
from abc import ABC, abstractmethod
from pymongo.database import Database
from typing import Any
class MongoDBService(ABC):
    """
    Service class for interacting with MongoDB to retrieve store items.
    """

    __client: MongoClient[Any]
    _db: Database[Any]

    def __init__(self, database_name: str = 'Store'):
        """
        Initializes the MongoDB client and selects the database and collection.
        """
        mongo_uri = os.getenv('MONGO_URI', 'mongodb://localhost:27017/')
        self.__client = MongoClient(mongo_uri)
        self._db = self.__client[database_name]
